tests/infrastructure/test_kubernetes_job.py

```diff
@@ -94,6 +101,16 @@ def get_batch_client(_):
     return mock
 
 
+@pytest.fixture
+def mock_k8s_batch_client_404(mock_k8s_batch_client):
+    mock_k8s_batch_client.delete_namespaced_job.side_effect = [ApiException(status=404)]
+
+
+@pytest.fixture
+def mock_k8s_batch_client_other_api_error(mock_k8s_batch_client):
+    mock_k8s_batch_client.delete_namespaced_job.side_effect = [ApiException(status=200)]
```

Comment:
Might want to choose a code other than a 200

Reply:
It seemed a bit funny last night, but in hindsight probably a bit confusing - fixed!